Mathlib/MeasureTheory/Integral/PathIntegral/Basic.lean

```diff
@@ -0,0 +1,489 @@
+
+theorem pathIntegrable_segment [NormedSpace ℝ E] :
+    PathIntegrable ω (.segment a b) ↔
+      IntervalIntegrable (fun t ↦ ω (lineMap a b t) (b - a)) volume 0 1 := by
+  rw [PathIntegrable, intervalIntegrable_congr]
+  rw [uIoc_of_le zero_le_one]
+  exact .mono Ioc_subset_Icc_self fun _t ↦ pathIntegralFun_segment ω a b
+
+theorem pathIntegral_segment [NormedSpace ℝ E] [NormedSpace ℝ F] (ω : E → E →L[𝕜] F) (a b : E) :
+    ∫ᵖ x in .segment a b, ω x = ∫ t in (0)..1, ω (lineMap a b t) (b - a) := by
+  rw [pathIntegral_def]
+  refine intervalIntegral.integral_congr fun t ht ↦ ?_
+  rw [uIcc_of_le zero_le_one] at ht
+  exact pathIntegralFun_segment ω a b ht
+
+@[simp]
+theorem pathIntegral_segment_const [NormedSpace ℝ E] [CompleteSpace F] (ω : E →L[𝕜] F) (a b : E) :
+    ∫ᵖ _ in .segment a b, ω = ω (b - a) := by
+  letI : NormedSpace ℝ F := .restrictScalars ℝ 𝕜 F
+  simp [pathIntegral_segment]
+
+/-- If `‖ω z‖ ≤ C` at all points of the segment `[a -[ℝ] b]`,
+then the path integral `∫ᵖ x in .segment a b, ω x` has norm at most `C * ‖b - a‖`. -/
+theorem norm_pathIntegral_segment_le [NormedSpace ℝ E] {C : ℝ} (h : ∀ z ∈ [a -[ℝ] b], ‖ω z‖ ≤ C) :
+    ‖∫ᵖ x in .segment a b, ω x‖ ≤ C * ‖b - a‖ := calc
+  ‖∫ᵖ x in .segment a b, ω x‖ ≤ C * ‖b - a‖ * |1 - 0| := by
+    letI : NormedSpace ℝ F := .restrictScalars ℝ 𝕜 F
+    rw [pathIntegral_segment]
+    refine intervalIntegral.norm_integral_le_of_norm_le_const fun t ht ↦ ?_
+    rw [segment_eq_image_lineMap] at h
+    rw [uIoc_of_le zero_le_one] at ht
+    apply_rules [(ω _).le_of_opNorm_le, mem_image_of_mem, Ioc_subset_Icc_self]
+  _ = C * ‖b - a‖ := by simp
+
+/-- If a 1-form `ω` is continuous on a set `s`,
+then it is path integrable along any $C^1$ path in this set. -/
+theorem ContinuousOn.pathIntegrable_of_contDiffOn [NormedSpace ℝ E] {s : Set E}
+    (hω : ContinuousOn ω s) (hγ : ContDiffOn ℝ 1 γ.extend I) (hγs : ∀ t, γ t ∈ s) :
+    PathIntegrable ω γ := by
+  apply ContinuousOn.intervalIntegrable_of_Icc zero_le_one
+  simp only [funext (pathIntegralFun_def ω γ)]
+  apply ContinuousOn.clm_apply
+  · exact hω.comp (by fun_prop) fun _ _ ↦ hγs _
+  · exact hγ.continuousOn_derivWithin uniqueDiffOn_Icc_zero_one le_rfl
+
+end PathOperations
+
+/-!
+### Algebraic operations on the 1-form
+-/
+
+section Algebra
+
+variable {𝕜 E F : Type*} [RCLike 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  [NormedAddCommGroup F] [NormedSpace 𝕜 F] {a b : E}
+  {ω ω₁ ω₂ : E → E →L[𝕜] F} {γ : Path a b} {t : ℝ}
+
+@[simp]
+theorem pathIntegralFun_add :
+    pathIntegralFun (ω₁ + ω₂) γ = pathIntegralFun ω₁ γ + pathIntegralFun ω₂ γ := by
+  ext; simp [pathIntegralFun]
+
+protected nonrec theorem PathIntegrable.add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    PathIntegrable (ω₁ + ω₂) γ := by
+  simpa [PathIntegrable] using h₁.add h₂
+
+theorem pathIntegral_add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    pathIntegral (ω₁ + ω₂) γ = (∫ᵖ x in γ, ω₁ x) + ∫ᵖ x in γ, ω₂ x := by
```

Comment:
use notation also on the left?

Also, could you fix priorities so that you don't need parentheses on the right?